### ISSUE TITLE
Install hangs at webpack

### ISSUE THREAD

**Issue by RitwikGupta** [jupyterlab/jupyterlab-latex#61]
When installing from either the egg or source, install is stuck at the `yarn run` command of `webpack --config webpack.prod.config.js` forever.

When attempting to manually run the command in the build script, the following error is thrown:

```text
node /home/ritwik/.pyenv/versions/3.6.4/lib/python3.6/site-packages/jupyterlab/staging/yarn.js run webpack --config webpack.prod.config.js
yarn run v1.3.2
$ /mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/.bin/webpack --config webpack.prod.config.js
/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/webpack.js:237
                                throw err;
                                ^

Error: Cannot find module '/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/webpack.prod.config.js'
    at Function.Module._resolveFilename (module.js:538:15)
    at Function.Module._load (module.js:468:25)
    at Module.require (module.js:587:17)
    at require (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at WEBPACK_OPTIONS (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/convert-argv.js:133:13)
    at requireConfig (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/convert-argv.js:135:6)
    at /mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/convert-argv.js:142:17
    at Array.forEach (<anonymous>)
    at module.exports (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/convert-argv.js:140:15)
    at yargs.parse (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/webpack.js:234:39)
    at Object.parse (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/yargs/yargs.js:539:18)
    at /mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/webpack.js:212:8
    at Object.<anonymous> (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack-cli/bin/webpack.js:504:3)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/mnt/c/Users/Ritwik/Documents/GitHub/jupyterlab-latex/node_modules/webpack/bin/webpack.js:12:2)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Function.Module.runMain (module.js:684:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Of note, I had to manually install webpack and webpack-cli because those were not installed with npm or yarn.

**Comment by ian-r-rose** [comment]
`PDF.js`, on which this extension relies, is a large and complex library, and including it makes the webpack build take significantly longer than it is otherwise (for me it takes almost a minute). How long did you wait for the build?

We would like to be able to split the pdf.js part of the build into a separate chunk, and the newly released webpack 4 is supposed to be faster, so there are some performance improvements that are possible.

**Comment by RitwikGupta** [comment]
I was waiting for approximately a bit over 4 hours on that command. For reference, I am building on WSL, but node, webpack, yarn, etc. do not have any compatibility problems individually with WSL that I know of.

**Comment by ian-r-rose** [comment]
Ah, well that certainly seems like it is hanging :)
I'll try rebooting into windows and see if I can reproduce.

**Comment by RitwikGupta** [comment]
Update: inside the `package.json` in `~/.pyenv/versions/3.5.5/lib/python3.5/site-packages/jupyterlab/staging`, I replaced `"build:prod": "webpack --config webpack.prod.config.js"` with `"build:prod": "webpack"`. The install works now! So it's not an extension issue.

**Comment by mpacer** [comment]
Do we have a way to migrate issues between repos on jupyterlab? I know @Carreau's meseeks bot could do it, but this should probably be closed here then.

**Comment by RitwikGupta** [comment]
I was waiting on closing the issue for exactly that reason. If there's a way to migrate it, please let me know how to do so (or if I should file a new issue). Otherwise, I can close it! 😃

**Comment by ian-r-rose** [comment]
I don't know how invoke meseeks to migrate the issue, but we can try to tag in @blink1073 to see if he has any ideas about what is going on here. But we should not be making you have to muck around in the `package.json` just to get an extension installed :)

**Comment by RitwikGupta** [comment]
For what it's worth, this issue does not occur when installing jupyterlab from an egg, but it does happen when installing jupyterlab from source. Maybe a Webpack/Yarn/WSL/config issue?

**Comment by blink1073** [comment]
Hi @RitwikGupta, can you share what commands you used to install both `jupyterlab` and this extension?  I can't figure out why this might be happening from that traceback.

**Comment by RitwikGupta** [comment]
@blink1073 Certainly. For this traceback, I installed Jupyterlab on Python 3.6.4 with `pip install jupyterlab`. This installed without any problems from the egg on PyPi. To install the extension, I ran the commands on the README, namely:
`pip install jupyterlab_latex`
`jupyter labextension install @jupyterlab/latex`

These commands hang wherever the install script hits
```
yarn run v1.3.2
$ webpack --config webpack.prod.config.js
```
The webpack command hangs indefinitely and **suppresses all output**. To see if I could get a traceback, I manually ran `node /home/ritwik/.pyenv/versions/3.6.4/lib/python3.6/site-packages/jupyterlab/staging/yarn.js run webpack --config webpack.prod.config.js` from the Jupyterlab staging folder. This is where you will get this traceback.

Running on Windows Subsystem for Linux
![image](https://user-images.githubusercontent.com/4959141/36939927-6a67baf4-1f07-11e8-9425-9c405555b6c4.png)

**Comment by blink1073** [comment]
Hi @RitwikGupta, does it work with `jupyter labextension install @jupyterlab/latex --dev` (which avoids the production build).

**Comment by RitwikGupta** [comment]
@blink1073 Yes, it does! The command `webpack` works, but `webpack --config webpack.prod.config.js` does not.

**Comment by cemulate** [comment]
I'm having the same issue with `jupyterlab` version `0.31.12`

Even with no extensions installed, `jupyter lab build` tries to run `webpack --config webpack.prod.config.js` and hangs for a very long time, while `jupyter lab build --dev` just runs `webpack` and finishes in a few seconds.

So it seems to be the production config that's causing a problem.

**Comment by cemulate** [comment]
There's a thread about this at jupyterlab/jupyterlab#4276

**Comment by jnvandermeer** [comment]
Installation in a fresh python=3 conda environment hangs on this issue.
(i.e. conda create - tex python=3 + source activate tex)

I tried the --dev option as outlined above, but no dice.
In addition I changed the "build" line in package.json as outlined above in the directory:
~/.conda/envs/tex/lib/python3.6/site-packages/jupyterlab/staging, but also didn't do much.

How long should it take to build the @jupyterlab/latex@file:../extensions/jupyterlab-latex-0.3.1.tgz?

Thanks!
J
